src/wallet/wallet2.h

```diff
@@ -673,16 +674,32 @@ namespace tools
     struct unsigned_tx_set
     {
       std::vector<tx_construction_data> txes;
-      std::pair<size_t, wallet2::transfer_container> transfers;
-      std::pair<size_t, std::vector<wallet2::exported_transfer_details>> new_transfers;
+      std::tuple<uint64_t, uint64_t, wallet2::transfer_container> transfers;
+      std::tuple<uint64_t, uint64_t, std::vector<wallet2::exported_transfer_details>> new_transfers;
 
       BEGIN_SERIALIZE_OBJECT()
-        VERSION_FIELD(1)
+        VERSION_FIELD(2)
         FIELD(txes)
-        if (version >= 1)
-          FIELD(new_transfers)
-        else
-          FIELD(transfers)
+        if (version == 0)
+        {
+          std::pair<size_t, wallet2::transfer_container> old_transfers;
+          FIELD(old_transfers);
+          std::get<0>(transfers) = std::get<0>(old_transfers);
+          std::get<1>(transfers) = std::get<0>(old_transfers) + std::get<1>(old_transfers).size();
+          std::get<2>(transfers) = std::get<1>(old_transfers);
+          return true;
+        }
+        if (version == 1)
+        {
+          std::pair<size_t, std::vector<wallet2::exported_transfer_details>> old_new_transfers;
+          FIELD(old_new_transfers);
```

Comment:
This is only written as JSON, not read. Read/write is binary, no keys. I'm totally fine returning false for versions < 2 if you'd rather. Since you're supposed to use the same version on both sides anyway.